### PR TITLE
Refact useFormattedYml

### PR DESCRIPTION
**Pull request by moczolaszlo**

### CODE REVIEW

File: source/javascripts/hooks/useFormattedYml.ts [hunk 1]
```diff
@@ -1,52 +1,17 @@
...
-    identityParser,
-  );
-
-  // NOTE: call function isn't referentially stable
-  useEffect(() => {
-    formatAppConfigRef.current = call;
+import { BitriseYml } from '../core/models/BitriseYml';
+
+const useFormattedYml = (appConfig: BitriseYml) => {
+  return useQuery({
```
Comment (AndrasEszes) [inline source/javascripts/hooks/useFormattedYml.ts:6]: I don't recommend using `useQuery` for POST requests. In my opinion, `useMutation` is better for this case.

**Review by AndrasEszes — DISMISSED** [review]

File: source/javascripts/components/ConfigurationYmlSource/ConfigurationYmlSourceDialog.tsx [hunk 2]
```diff
@@ -63,7 +61,7 @@ type ConfigurationYmlSourceDialogProps = {
   isOpen: boolean;
   onClose: () => void;
   initialUsesRepositoryYml: boolean;
-  appConfig: AppConfig | string;
+  appConfig: string;
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/components/ConfigurationYmlSource/ConfigurationYmlSourceDialog.tsx:64]: Will it be a string always?

File: source/javascripts/components/UpdateConfigurationDialog/UpdateConfigurationDialog.tsx [hunk 3]
```diff
@@ -46,7 +47,7 @@ const UpdateConfigurationDialog = (props: UpdateConfigurationDialogProps) => {
     }
   };
 
-  const yml = useFormattedYml(appConfig);
+  const ymlString = useFormattedYml(appConfig as BitriseYml) || '';
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/components/UpdateConfigurationDialog/UpdateConfigurationDialog.tsx:50]: Is it a BitriseYml always?

File: source/javascripts/components/YmlEditorHeader/YmlEditorHeader.tsx [hunk 4]
```diff
@@ -154,7 +154,7 @@ const YmlEditorHeader = (props: YmlEditorHeaderProps) => {
         isOpen={isOpen}
         onClose={onClose}
         initialUsesRepositoryYml={usesRepositoryYml}
-        appConfig={appConfig}
+        appConfig={appConfig as string}
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/components/YmlEditorHeader/YmlEditorHeader.tsx:157]: Are you sure its not a mistake to cast it to a string?

**Review by AndrasEszes — APPROVED** [review]

**Pull request merged**